zerver/tests/test_bugdown.py

```diff
@@ -1892,3 +1896,267 @@ def test_avatar_of_unregistered_user(self) -> None:
+                                "type": "string"
+                            },
+                            "required": True
+                        }
+                    ]
+                },
+                "brew": {}  # the data is irrelevant as is should be rejected.
+            }
+        }
+    }
+
+    spec_mock_using_object_and_array_with_example = {
+        "paths": {
+            "/endpoint": {
+                "get": {
+                    "description": "Get some info.",
+                    "parameters": [
+                        {
+                            "name": "param1",
+                            "in": "path",
+                            "description": "An object",
+                            "schema": {
+                                "type": "object"
+                            },
+                            "example": {
+                                "thing1": "something",
+                                "thing2": 10
+                            },
+                            "required": True
+                        },
+                        {
+                            "name": "param2",
+                            "in": "path",
+                            "description": "An array",
+                            "schema": {
+                                "type": "array"
+                            },
+                            "example": [
+                                "thing1",
+                                "thing2"
+                            ],
+                            "required": True
+                        }
+                    ]
+                }
+            }
+        }
+    }
+
+    spec_mock_using_object_without_example = {
+        "paths": {
+            "/endpoint": {
+                "get": {
+                    "description": "Get some info.",
+                    "parameters": [
+                        {
+                            "name": "param1",
+                            "in": "path",
+                            "description": "An object",
+                            "schema": {
+                                "type": "object"
+                            },
+                            "required": True
+                        }
+                    ]
+                }
+            }
+        }
+    }
+
+    spec_mock_using_array_without_example = {
+        "paths": {
+            "/endpoint": {
+                "get": {
+                    "description": "Get some info.",
+                    "parameters": [
+                        {
+                            "name": "param1",
+                            "in": "path",
+                            "description": "An array",
+                            "schema": {
+                                "type": "array"
+                            },
+                            "required": True
+                        }
+                    ]
+                }
+            }
+        }
+    }
```

Comment:
What's the thinking behind including these partial OpenAPI data structures, over just using a specific endpoint from `zulip.yaml` that uses the appropriate HTTP method (etc.)?  I feel like it wouldn't be hard to do, and the test would be more end-to-end.  (And I don't expect us to change the API itself so often that doing updates to this test when we do so would be expensive)

Reply:
I had two reasons:
1. To be able to define the exact condition I wanted to test easily instead of having to search for an appropriate endpoint.
2. I originally didn't want changes in the actual documentation to affect this test.

I'll make the change though :+1:

Reply:
I still have to keep a few (e.g. for the cases where an example is missing and would generate an exception). But I've been able to remove the other data structures.

Reply:
Great